src/source/mod.rs

```diff
@@ -333,6 +335,13 @@ where
         skippable::skippable(self)
     }
 
+    fn trackable(self) -> TrackPosition<Self>
```

Comment:
I like the name `TrackPosition`, it communicates that it has to do with the position of the source and tracking it. Would you be okay with changing the name `trackable` to `track_position`? Then the factory function clearly communicates that what is tracked is the position.